manual_flagging/flagging.py

```diff
@@ -0,0 +1,517 @@
+    be returned with the same value. For example, if res and condos have the 
+    same column groupings, joining the group column from sale.flag to sale.group_mean
+    by 'group' and 'run_id' could potentially return two groups. This market type
+    append fixes that. This is also added in the group_mean data.
+    """
+    market_value = df["market"]
+    df_copy["group"] = df_copy["group"].astype(str) + "-" + market_value
+
+    # Add the edited or unedited dataframe to the new dictionary
+    dfs_to_finalize[df_name]["df"] = df_copy
+
+# - - - - - - -
+# Finalize data to write and create data for all metadata tables
+# - - - - - - - -
+
+if inputs["manual_update"] == True:
+    # Group the existing data by 'ID' and find the maximum 'version' for each sale
+    existing_max_version = (
+        df_flag_table.groupby("meta_sale_document_num")["version"]
+        .max()
+        .reset_index()
+        .rename(columns={"version": "existing_version"})
+    )
+
+
+dfs_to_finalize_list = [details["df"] for details in dfs_to_finalize.values()]
+df_to_finalize = pd.concat(dfs_to_finalize_list, axis=0)
+
+df_to_write, run_id, timestamp = flg.finish_flags(
+    df=df_to_finalize, start_date=inputs["time_frame"]["start"]
+)
+
+if inputs["manual_update"] == True:
+    # Merge, compute new version, and drop unnecessary columns
+    df_to_write = (
+        df_to_write.merge(existing_max_version, on="meta_sale_document_num", how="left")
+        .assign(
+            version=lambda x: x["existing_version"]
+            .apply(lambda y: y + 1 if pd.notnull(y) else 1)
+            .astype(int)
+        )
+        .drop(columns=["existing_version"])
+    )
+
+run_filter = str(
+    {"housing_market_type": inputs["housing_market_type"], "run_tri": inputs["run_tri"]}
+)
+
+# Get sale.parameter data
+df_parameter = flg.get_parameter_df(
+    df_to_write=df_to_write,
+    df_ingest=df_ingest,
+    run_filter=run_filter,
+    iso_forest_cols=inputs["iso_forest"],
+    stat_groups=inputs["stat_groups"],
+    tri_stat_groups=inputs["tri_stat_groups"],
+    housing_market_class_codes=inputs["housing_market_class_codes"],
+    dev_bounds=inputs["dev_bounds"],
+    ptax_sd=inputs["ptax_sd"],
+    rolling_window=inputs["rolling_window_months"],
+    date_floor=inputs["time_frame"]["start"],
+    short_term_thresh=flg_model.SHORT_TERM_OWNER_THRESHOLD,
+    min_group_thresh=inputs["min_groups_threshold"],
+    run_id=run_id,
+)
+
+# Standardize dtypes to prevent Athena errors
+df_parameter = flg.modify_dtypes(df_parameter)
+
+# Get sale.group_mean data
+df_group_means = []  # List to store the transformed DataFrames
+
+for df_name, df in dfs_to_finalize.items():
+    df_group_mean = flg.get_group_mean_df(
+        df=dfs_flagged[df_name]["df"],
+        stat_groups=df["columns"],
+        run_id=run_id,
+        condos=df["condos_boolean"],
+    )
+    market_value = df["market"]
+    df_group_mean["group"] = df_group_mean["group"].astype(str) + "-" + market_value
```

Comment:
Tried this for about 2 hours and got stumped. It requires a decent amount of changes in the flagging pipeline so I'm thinking about skipping it for now. It might be worth making an issue about this though.